File: euler1.py
from functools import reduce

def solve(max):
    nums = [num for num in range(1, max) if num % 3 == 0 or num % 5 == 0]
    sum = reduce(lambda x, y: x + y, nums)
    return sum

print(solve(10))
print(solve(1000))

File: euler1.c
#include <stdio.h>

// Declaration
int solve(int max);

int solve(int max){
    int sum = 0;
    int i = 0;
    for(i = 0; i < max; i++){
        if(i % 3 == 0 || i % 5 == 0){
            sum += i;
        }
    }

    return sum;
}

int main(int argc, char *argv[]){
    printf("%d\n", solve(10));
    printf("%d\n", solve(1000));
    return 0;
}
